packages/web/src/app/Footer.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import { DialogRef } from '../shared/components/Dialog';
+import { Icon } from '../shared/components/Icon';
+import LanguageDialog from './LanguageDialog';
+import interfaceLanguages from './languages.json';
+
+export default function Footer() {
+  const i18nextLanguage = localStorage.getItem('i18nextLng');
+
+  const languageDialog = useRef<DialogRef>(null);
+  const { t, i18n } = useTranslation();
+
+  useEffect(() => {
+    if (i18nextLanguage == null) {
+      // This never occurs.
+      languageDialog.current?.open();
+    }
```

Comment:
> i18next already used a local storage property. If not defined, it will make an educated guess. Before i18next loads we can check ourselves, and show the modal if the local storage prop isn't defined. Then the modal will default to i18next's guess

I'm not sure how to access local storage before i18next

Reply:
Even from `main.tsx`, that field is always set in local storage before the component function is called.

Reply:
If we add this line to `i18next` before we call `init` then we can check local storage first:
```ts
export const initialLanguageChosen = !!localStorage.getItem('i18nextLng');
```

Now we can import that into the footer to show the modal. React refs don't need to be added to the dependency list of an effect, so we can guarantee that this effect only runs on first render
```ts
useEffect(() => {
  if (!initialLanguageChosen) {
    languageDialog.current?.open();
  }
}, []);
```